### DISCUSSION TITLE
How to pass compilation options to clang-tidy in VScode

### DISCUSSION THREAD

**Discussion by fonsecapaulo** [microsoft/vscode-cpptools#9748] [Q&A]
Hi all,
What is the proper way to pass compilation options to clang-tidy in Vscode using the native C++ extension?

I am specially interested in -std=c++17

I tried : 

```json
"C_Cpp.codeAnalysis.clangTidy.args": [
    "--",
    "-std=c++17"
],
```

```json
"C_Cpp.codeAnalysis.clangTidy.args": [
   "-std=c++17"
],
```

but none of them seem to work.

I would like this to be picked up when i trigger clang-tidy with: __Run Code Analysis on active file__

**Comment by sean-mcmanus** [comment]
By default, it should pick up the std value from the IntelliSense configuration, e.g. the cppStandard in c_cpp_properties.json or the equivalent value via other configuration methods. Is that sufficient?
